Sources/lexicon-generate/CodeGenerator.swift

```diff
@@ -0,0 +1,105 @@
+import ArgumentParser
+import Collections
+import Foundation
+import Lexicon
+import LexiconGenerators
+
+@main
+struct CodeGeneratorCommand: AsyncParsableCommand {
+
+	static var configuration = CommandConfiguration(
+		commandName: "lexicon-generate",
+		abstract: "A utility for generating code from lexicon documents.",
+		version: "1.0.0"
+	)
+
+	@Argument(help: "File path or URL to the lexicon")
+	var input: URL
+
+	@Option(
+		name: .shortAndLong,
+		help: "Output path excluding extension, if not specified the same directory and name of the lexicon will be used"
+	)
+	var output: URL?
+
+	@Option(
+		name: .shortAndLong,
+		help:
+		"""
+		Types of code to generate. Comma separated.
+
+		Generators:
+			\(Lexicon.Graph.JSON.generators.commandHelp)
+
+		Example:
+			--type swift,kotlin
+		"""
+	)
+	var type: [String]
+
+	@Flag(name: .shortAndLong)
+	var quiet: Bool = false
+
+	private var isLogging: Bool { !quiet }
+
+	mutating func run() async throws {
+		let name = String(input.lastPathComponent.split(separator: ".")[0])
+		if isLogging {
+			print("\(name) lexicon")
+		}
+		let lexicon = try await Lexicon.from(
+			TaskPaper(Data(contentsOf: input)).decode()
+		)
+		let json = await lexicon.json()
+		let code = try type.map { command in
+			guard let generator = Lexicon.Graph.JSON.generators.find(command) else {
+				fatalError("Unable to find a generator for \(command)")
```

Comment:
Just checking because I'm not too familiar with Swift + Command Line, do you know if there is difference between calling `fatalError` and `exit(0)`?

Reply:
There is a difference yes, `exit(0)` is deemed as a successful exit, which is the default if you don't error out. Anything other than code `0` for an exit status is considered an error - in this case, fatalError produces a non-zero with a message for the consumer to understand the reason

Reply:
Brill, thanks. 
I was wondering whether other scripts that may eventually call `lexicon-generate` would be able to detect a failure in the same way with `fatalError` and `exit`, but sounds like they can!